prog/test/postgres_resource.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Test::PostgresResource < Prog::Test::Base
+  semaphore :destroy
+
+  def self.assemble
+    postgres_service_project = Project.create(name: "Postgres-Service-Project") { _1.id = Config.postgres_service_project_id }
+    postgres_service_project.associate_with_project(postgres_service_project)
+
+    postgres_test_project = Project.create(name: "Postgres-Test-Project")
+    postgres_test_project.associate_with_project(postgres_test_project)
+
+    Strand.create_with_id(
+      prog: "Test::PostgresResource",
+      label: "start",
+      stack: [{
+        "postgres_service_project_id" => postgres_service_project.id,
+        "postgres_test_project_id" => postgres_test_project.id
+      }]
+    )
+  end
+
+  label def start
+    hop_create_postgres_resource
+  end
+
+  label def create_postgres_resource
+    st = Prog::Postgres::PostgresResourceNexus.assemble(
+      project_id: frame["postgres_test_project_id"],
+      location: "hetzner-fsn1",
+      name: "postgres-test-standard",
+      target_vm_size: "standard-2",
+      target_storage_size_gib: 128,
+      version: "17",
+      flavor: "standard",
+      ha_type: "none"
+    )
+
+    update_stack({"postgres_resource_id" => st.id})
+    hop_wait_postgres_resource
+  end
+
+  label def wait_postgres_resource
+    if postgres_resource.strand.label == "wait" &&
+        postgres_resource.representative_server.run_query("SELECT 1") == "1"
+      hop_test_basic_connectivity
+    else
+      nap 10
+    end
+  end
+
+  label def test_basic_connectivity
```

Comment:
Just one label that tests everything is enough IMO. I think in a regular scenario, either all these test fail or all these tests pass. We don't gain any extra information with separate tests.

Let's have one label, that creates tables, loads data and run queries. In fact, we can have one sql file that does all these and we can only run that.